### ISSUE TITLE
Data loader does not check that segmentation labels are mutually exclusive

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/InnerEye-DeepLearning#339]
`load_labels_from_dataset_source` builds up the ground truth tensor from individual binary masks. It sets the background mask correctly, but does not check if each pixel is really only in 1 class. If that assumption is violated, model training may later fail when computing the FocalLoss.
Suggestion:
- Add a check in `load_labels_from_dataset_source`. If violated, it should print out the actionable information about how to fix that: Which classes overlap, how many pixels overlap, what are the file names?
- we have a script "check annotation quality" that can help diagnose. Can we add that as an optional part of a training run, with a flag `check_labels`? What might be tricky: at the moment, that script takes a structure name as an argument. We would want all pairs of structures checked for overlap, but we don't want a quadratic runtime.

[AB#3908](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/3908)

**Comment by ant0nsc** [comment]
- One model where that problem happens is the Lung model
- normal codepath: Raise an exception if the labels are not mutually exclusive. 
- there should be a commandline switch to not execute the check for mutually exclusive labels.
- The error message raised if something is wrong in `load_labels_from_dataset_source` should suggest to trigger the check annotation quality.

**Comment by ant0nsc** [comment]
- When using FocalLoss, the segmentations need to be mutually exclusive.
- when using SoftDice loss, they don't need to be (the loss function will cope, but the models assume that labels are mutually exclusive because we do a softmax)
- Maybe we want to have a flag that turns the check for mutually exclusive segmentations on/off? If people use FocalLoss, they must do the check. If SoftDice, not having mutually exclusive labels is at your own risk.

**Comment by ant0nsc** [comment]
For tests: Could test with FocalLoss in isolation, just feed in tensors of appropriate size. 
End-to-end-test: maybe write something like `test_runner1`, but with a segmentation model. 
Check is probably best in InnerEyeContainer setup method
